tests/lora/test_lora_layers_peft.py

```diff
@@ -1392,6 +1392,64 @@ class StableDiffusionXLLoRATests(PeftLoraLoaderMixinTests, unittest.TestCase):
     }
 
 
+class UNet2DConditionModelLoRATests(unittest.TestCase):
+    def get_dummy_components(self):
+        unet_kwargs = {
+            "block_out_channels": (32, 64),
+            "down_block_types": ("CrossAttnDownBlock2D", "DownBlock2D"),
+            "up_block_types": ("UpBlock2D", "CrossAttnUpBlock2D"),
+            "cross_attention_dim": 32,
+            "attention_head_dim": 8,
+            "out_channels": 4,
+            "in_channels": 4,
+            "layers_per_block": 2,
+            "sample_size": 32,
+        }
+        torch.manual_seed(0)
+        unet = UNet2DConditionModel(**unet_kwargs)
+        unet_lora_config = LoraConfig(
+            r=4, lora_alpha=4, target_modules=["to_q", "to_k", "to_v", "to_out.0"], init_lora_weights=False
+        )
+        return unet, unet_lora_config
+
+    def get_dummy_inputs(self):
+        batch_size = 2
+        num_channels = 4
+        sizes = (32, 32)
+
+        noise = floats_tensor((batch_size, num_channels) + sizes).to(torch_device)
+        time_step = torch.tensor([10]).to(torch_device)
+        encoder_hidden_states = floats_tensor((batch_size, 4, 32)).to(torch_device)
+
+        return {"sample": noise, "timestep": time_step, "encoder_hidden_states": encoder_hidden_states}
+
+    def test_inference(self):
+        unet, unet_lora_config = self.get_dummy_components()
+        unet = get_peft_model(unet, unet_lora_config)
+        inputs = self.get_dummy_inputs()
+        outputs = unet(**inputs).sample
+
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            unet.save_pretrained(tmpdirname)
+            self.assertTrue(os.path.isfile(os.path.join(tmpdirname, "adapter_model.bin")))
+            self.assertTrue(os.path.isfile(os.path.join(tmpdirname, "adapter_config.json")))
+
+            # Replicate training.
+            state_dict = torch.load(os.path.join(tmpdirname, "adapter_model.bin"), map_location="cpu")
+            state_dict = {k: torch.randn_like(v) for k, v in state_dict.items()}
+            torch.save(state_dict, os.path.join(tmpdirname, "adapter_model.bin"))
```

Comment:
@BenjaminBossan, I did try to specify `init_lora_weights` to True above, but that led to a test failure. 

Any suggestions?

Reply:
You mean in this line?

https://github.com/huggingface/diffusers/pull/5958/files#diff-8175616511c2a255aaaf878f150c8a33b2789722fe7bd7434c5f16d3dc967a46R1411

I don't know what the failure was but I assume it's in the last line that checks that `outputs != outputs_with_lora`. The thing that is happening here is: If we set `init_lora_weights=True` (or `"gaussian"` in the future), the LoRA weights will be initialized such that they perform an identity operation (`lora_B` will be all zeros).

For context, the `init_lora_weights=False` option was added exactly with the intent to get a non-identity transform, so that we can see the effect of LoRA being added without the need to train the model first, so it's basically a debug option.

If we want to have tests for different settings of `init_lora_weights`, those should be separate tests that don't check the model output (as that doesn't change) but inspect the LoRA weights instead.

Reply:
Yeah I would have guessed that. Will investigate more as to why specifying `init_lora_weights` to `False` doesn't cut it for more and instead, I am having to explicitly emulate training (by randomizing the weights in the state dict).

Reply:
Sorry. I mean, when I initialise the LoRaConfig object with `init_lora_weights` set to `False`, the assertion test doesn't pass.

Reply:
Ah I see, that's indeed strange. Since the CI errors before that, I had assumed that the last assertion would pass.

Reply:
Sorry, it was a mistake on my part. Have fixed that.